### PR TITLE
Migrate OTel exporter tests to TestProxy

### PR DESCRIPTION
**Pull request by srnagar**
This PR migrates the QuickPulseIntegrationTest recordings to TestProxy.

### CODE REVIEW

**Review by samvaity — APPROVED** [review]
🎉

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/AppConfigurationExporterIntegrationTest.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ public class AppConfigurationExporterIntegrationTest extends MonitorExporterClie
     @BeforeEach
     public void setupTest(TestInfo testInfo) {
         Assumptions.assumeFalse(getTestMode() == TestMode.PLAYBACK, "Skipping playback tests");
+        super.setupTest(testInfo);
```
Comment (samvaity) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/AppConfigurationExporterIntegrationTest.java:44]: why do we need this?

**Review by trask — APPROVED** [review]

**Comment by trask** [comment]
@heyams @jeanbisutti the docs for how to use the new framework are here: https://github.com/Azure/azure-sdk-for-java/wiki/Test-Proxy-Migration

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/AppConfigurationExporterIntegrationTest.java (hunk 1 above)
Comment (srnagar) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/AppConfigurationExporterIntegrationTest.java:49]: This is needed for the setup to be done (i.e. call the setupTest method of the base class) when the test mode is not PLAYBACK.

**Pull request merged**